src/analysis/refactor_open.ml

```diff
@@ -1,36 +1,52 @@
 open Std
 
-let qual_or_unqual_path mode leftmost_ident path p =
+(** [leftmost_lident lid] returns the leftmost part of [lid], e.g., 
+    given [String.Map.empty], [String] is returned *)
+let rec leftmost_lident (lid : Longident.t) = 
+  match lid with
+  | Lident s -> s
+  | Ldot(lid', _) -> leftmost_lident lid'
+  | Lapply(_, _) -> raise @@ Invalid_argument "leftmost_lident: don't support Lapply"
```

Comment:
I think this could be
- named `head` (it's the equivalent of `Path.head`)
- moved to the `Longident` module (and we'd upstream it later)
- `raise @@ Invalid_argument ...` is `invalid_arg ...`. 
  But here I think we can just `assert false` (like `Path.head`).